tests/integration/test_relation.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import json
+from pathlib import Path
+
+import kubernetes
+import requests
+from pytest_operator.plugin import OpsTest
+
+INGRESS_APP_NAME = "ingress"
+ANY_APP_NAME = "any"
+NEW_HOSTNAME = "any.other"
+NEW_PORT = 18080
+
+
+async def test_build_and_deploy(ops_test: OpsTest, run_action):
+    """
+    arrange: no arrange.
+    act: build and deploy nginx-ingress-integrator charm, also deploy and relate an any-charm
+        application for test purposes.
+    assert: all application should be active.
+    """
+    ingress_lib = Path("lib/charms/nginx_ingress_integrator/v0/ingress.py").read_text()
+    any_charm_script = Path("tests/integration/any_charm.py").read_text()
+    any_charm_src_overwrite = {
+        "ingress.py": ingress_lib,
+        "any_charm.py": any_charm_script,
+    }
+
+    async def build_and_deploy():
+        charm = await ops_test.build_charm(".")
+        return await ops_test.model.deploy(
+            str(charm), application_name="ingress", series="focal", trust=True
+        )
+
+    await asyncio.gather(
```

Comment:
Should this be in a fixture?

Reply:
Here is deploying the Nginx ingress charm and any-charm which is unique to this test module and not applicable to other test modules. Also, it's running everything concurrently to speed up the building and deploying process which makes it difficult to separate into fixtures. That's why I use a test case instead of a fixture.